examples/aws-sagemaker/create_endpoint.py

```diff
@@ -0,0 +1,171 @@
+        self.model_name = model_name
+        self.initial_instance_count = instance_count
+        self.instance_type = instance_type
+        self.production_variants = [
+            {
+                "VariantName": self.variant_name,
+                "ModelName": self.model_name,
+                "InitialInstanceCount": self.initial_instance_count,
+                "InstanceType": self.instance_type,
+            }
+        ]
+        self.endpoint_config_name = config_name
+        self.endpoint_config = {
+            "EndpointConfigName": self.endpoint_config_name,
+            "ProductionVariants": self.production_variants,
+        }
+        self.endpoint_name = endpoint_name
+
+    def create_image(self):
+
+        subprocess.run(self.build_cmd, shell=True)
+
+    def create_ecr_repo(self):
+
+        try:
+            ecr = boto3.client("ecr", region_name=self.region_name)
+            ecr.create_repository(repositoryName=self.repo_name)
+
+        except ecr.exceptions.RepositoryAlreadyExistsException:
+            pass
+
+        repo_check = ecr.describe_repositories(repositoryNames=[self.repo_name])
+        pprint(repo_check["repositories"])
+
+    def push_image(self):
+
+        subprocess.call(["sh", self.push_script])
+
+    def create_model(self):
+
+        self.sm_boto3.create_model(
+            ModelName=self.model_name,
+            Containers=self.image,
+            ExecutionRoleArn=self.ROLE_ARN,
+            EnableNetworkIsolation=False,
+        )
+
+    def create_endpoint_config(self):
+
+        self.sm_boto3.create_endpoint_config(**self.endpoint_config)
+
+    def create_endpoint(self):
+
+        self.sm_boto3.create_endpoint(
+            EndpointName=self.endpoint_name,
+            EndpointConfigName=self.endpoint_config_name,
+        )
+
+        pprint(self.sm_boto3.describe_endpoint(EndpointName=self.endpoint_name))
+
+    def nuke_endpoint(self):
+
+        self.sm_boto3.delete_endpoint(EndpointName=self.endpoint_name)
+        self.sm_boto3.delete_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name
+        )
+        self.sm_boto3.delete_model(ModelName=self.model_name)
+
+        print("endpoint nuked!")
+
+
+if __name__ == "__main__":
+
+    Sage = SparseMaker(
+        instance_count=1,
+        region_name="us-east-1",
+        instance_type="ml.c5.large",
+        image_name="deepsparse-sagemaker-example",
+        repo_name="deepsparse-sagemaker",
+        image_push_script="./push_image.sh",
+        variant_name="QuestionAnsweringDeepSparseDemo",
+        model_name="question-answering-example",
+        config_name="QuestionAnsweringExampleConfig",
+        endpoint_name="question-answering-example-endpoint",
+        role_arn="<PLACEHOLDER>",
+    )
+
+    Sage.create_image()
+    Sage.create_ecr_repo()
+    Sage.push_image()
+    Sage.create_model()
+    Sage.create_endpoint_config()
+    Sage.create_endpoint()
+    # Sage.nuke_endpoint()
```

Comment:
if this is being run as a script we should have shutting down the endpoint as part of it. maybe once the endpoint is setup prompt to user to press a key to shutdown endpoint and stop the process. even better if this happened on exit/deconstruction